app/src/androidTest/java/com/github/sdp/mediato/ProfileFragmentTest.java

```diff
@@ -67,39 +76,76 @@ public void testUsername() {
+
+    // Check that the recycler view is initially empty
+    recyclerView.check(matches(hasItemCount(0)));
   }
 
-  /*// Test whether the "Add movie" button is displayed
+  // Test that an item is added to the collection on click of the AddMediaButton
   @Test
-  public void testAddMovieButton() {
-    ViewInteraction collectionView = onView(withId(R.id.layout_collection));
-    collectionView.check(matches(hasDescendant(withId(R.id.add_media_button))));
-    onView(withId(R.id.add_media_button)).check(matches(isDisplayed()));
+  public void testAddMediaButton() {
+    ViewInteraction collectionRecyclerView = onView(withId(R.id.collection_recycler_view));
+    ViewInteraction addMediaButton = onView(withId(R.id.add_media_button));
+
+    int initialItemCount = getRecyclerViewItemCount(R.id.collection_recycler_view);
+
+    collectionRecyclerView.check(matches(hasItemCount(initialItemCount)));
+    addMediaButton.perform(click());
+    collectionRecyclerView.check(matches(hasItemCount(initialItemCount + 1)));
   }
 
-  // Test whether the collection view is displayed
-  @Test
-  public void testHorizontalScrollList() {
-    ViewInteraction collectionView = onView(withId(R.id.collectionRecyclerView));
-    collectionView.check(matches(isDisplayed()));
+  /**
+   * A matcher to check if a RecyclerView has a certain amount of items.
+   *
+   * @param count the expected number of items.
+   * @return a Matcher to check if a view has count amount of items.
+   */
+  public static Matcher<View> hasItemCount(int count) {
+    return new BoundedMatcher<View, RecyclerView>(RecyclerView.class) {
+      @Override
+      protected boolean matchesSafely(RecyclerView recyclerView) {
+        return recyclerView.getAdapter().getItemCount() == count;
+      }
+
+      @Override
+      public void describeTo(Description description) {
+        description.appendText("RecyclerView with item count: " + count);
+      }
+    };
   }
 
-  // Test whether a movie item, which consists of an image of a movie and a title and rating text,
-  // is displayed and has the correct title and rating
-  @Test
-  public void testMovieItem() {
-    ViewInteraction movieItem = onView(withId(R.layout.layout_movie_item));
-    movieItem.check(matches(isDisplayed()));
-    movieItem.check(matches(hasDescendant(withText("Skyfall"))));
-    movieItem.check(matches(hasDescendant(withText("Rating: 10"))));
-  }*/
+  /**
+   * Returns the number of items in a RecyclerView hosted by the TestingActivity
+   *
+   * @param id the id of the RecyclerView
+   * @return the number of items currently in the RecyclerView
+   */
+  private int getRecyclerViewItemCount(int id) {
+    final int[] itemCount = new int[1];
 
+    scenario.onActivity(activity -> {
+      RecyclerView recyclerView = activity.findViewById(id);
+      itemCount[0] = recyclerView.getAdapter().getItemCount();
+    });
+    return itemCount[0];

```

Comment:
Why is an int array used here?

Reply:
Actually an `int` itself would do the job here. The problem is that inside the lambda expression I can only use final variables, and if I make `itemCount` a `final int` I cannot change it's value inside the lambda expression anymore. I'm just using the `int[]` array as a fix for this because I can declare it as final but still change the values of the array. I could do the same thing with a list but it's not relevant here because it does not really represent a list and is just used as a fix to modify a single `int` inside the lambda expression.